### PR TITLE
fix PartialInviteChannel.__str__

### LINKED ISSUE #382: PartialInviteChannel name can be None
### Summary

Attempting to cast a PartialInviteChannel's instance to a string will result in a TypeError for an unnamed group DM invite, because there is no channel name.

### Reproduction Steps

- Create an empty group DM, and don't give it a name yet (the title should say "Unnamed")
- Create an invite for it, and get the invite object with `client.fetch_invite`.
- The invite's channel's name should be None.

### Minimal Reproducible Code

_No response_

### Expected Results

The channel name should always be a string, as per https://docs.disnake.dev/en/latest/api.html?highlight=partialinvitechannel#disnake.PartialInviteChannel.name

### Actual Results

The channel's name with the steps above is None, which means the channel's `__str__` returns None, which makes functions like `print` error out: 
```py
TypeError: __str__ returned non-string (type NoneType)
```

### Intents

All of them but presence

### System Information

Linux, Python 3.9.10. Let me know if you can't reproduce it with the info provided.

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

One possible solution would be to just return an empty string. Another would be to return "Unnamed" which is what the Discord client displays in this case for the channel. Either way like the error says `__str__` should always return a string.

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

Due to some discord api inconsistencies (and a minor doc update, https://github.com/discord/discord-api-docs/pull/4571), this is a simple implementation to fix `__str__`.

A future implementation of PartialInviteChannel will properly use the recipients key to 
provide a close equivalent of how the discord client shows a group DM name.

Unfortunately, the api currently is unclear as what information is provided in the 
recipients key for group dms. The documentation states the array is full user objects,
which is not the case from testing. As such, this has not been implemented.

This can be tracked at discord/discord-api-docs#4572

- fix: mark channel name as both optional and nullable in payload typings
- fix: make PartialInviteChannel.__str__ always return a string

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint` or `pre-commit run --all-files`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

closes #382

### CODE REVIEW

**Review by shiftinv — APPROVED** [review]
lgtm, could you merge master into this branch?

**Review by EQUENOS — APPROVED** [review]
lgtm
This branch is outdated though

File: disnake/types/channel.py [hunk 1]
```diff
@@ -44,10 +44,10 @@ class PermissionOverwrite(TypedDict):
 
 class _BaseChannel(TypedDict):
     id: Snowflake
-    name: str
```
Comment (shiftinv) [inline disnake/types/channel.py:47]: Due to this change, the `ThreadChannel` and `GuildDirectory` types are missing a `name` field now

Comment (onerandomusername) [reply disnake/types/channel.py:47]: https://github.com/DisnakeDev/disnake/pull/383/commits/2095bf3379d830fdf8f30f3a3a0360b2809cbb74

**Pull request merged**